modules/juce_dsp/juce_dsp.h

```diff
@@ -58,7 +58,7 @@
 #include <juce_audio_basics/juce_audio_basics.h>
 #include <juce_audio_formats/juce_audio_formats.h>
 
-#if defined(_M_X64) || defined(__amd64__) || defined(__SSE2__) || (defined(_M_IX86_FP) && _M_IX86_FP == 2)
+#if defined(_M_X64) || defined(__amd64__) || defined(__SSE2__) || (defined(_M_IX86_FP) && _M_IX86_FP == 2) || defined(__riscv) || defined(__EMSCRIPTEN__)

```

Comment:
Hi @falkTX, would you mind explaining how the change of this line works? I think on RISC-V, there is no header `<immintrin.h>`, and I did encounter the error:

```
In file included from /build/surge-xt/src/surge-xt/libs/sst/sst-filters/include-extras/sst/filters/FilterPlotter.h:7,                                                                                                                                                                                              
                 from /build/surge-xt/src/surge-xt/src/common/dsp/DSPExternalAdapterUtils.cpp:6:                                                                                                                                                                                                                   
/build/surge-xt/src/surge-xt/libs/JUCE/modules/juce_dsp/juce_dsp.h:74:12: fatal error: immintrin.h: No such file or directory                                                                                                                                                                                      
   74 |   #include <immintrin.h>                                            
      |            ^~~~~~~~~~~~~                                                                                                                                                                                                                                                                                   
compilation terminated.                                                                                                                                                                                                                                                                                            
make[2]: *** [src/common/CMakeFiles/surge-common.dir/build.make:314: src/common/CMakeFiles/surge-common.dir/dsp/DSPExternalAdapterUtils.cpp.o] Error 1                                                                                                                                                             
make[2]: *** Waiting for unfinished jobs....                                                                                                                                                                                                                                                                       
In file included from /build/surge-xt/src/surge-xt/src/common/dsp/effects/chowdsp/shared/chowdsp_DelayLine.h:28,                                                                                                                                                                                                   
                 from /build/surge-xt/src/surge-xt/src/common/dsp/effects/chowdsp/NeuronEffect.h:24,                                                                                                                                                                                                               
                 from /build/surge-xt/src/surge-xt/src/common/dsp/Effect.cpp:23:                                                                         
/build/surge-xt/src/surge-xt/libs/JUCE/modules/juce_dsp/juce_dsp.h:74:12: fatal error: immintrin.h: No such file or directory                                                                                                                                                                                      
   74 |   #include <immintrin.h>                                                                                                                                                                                                                                                                                   
      |            ^~~~~~~~~~~~~                                                                                                                                                                                                                                                                                   
compilation terminated.
```

Reply:
it works via simde which provides SSE-like implementations for non-SSE platforms.
to get things further you can setup a file like this https://github.com/DISTRHO/Cardinal/blob/main/include/simd-compat/immintrin.h so current codebases do not have to change much besides just allowing riscv builds to pass.

Reply:
that works too, and is indeed more clean.
but to make it proper I would suggest to instead of a macro for simde, alike this:

```
#elif defined(JUCE_USE_SIMDE)
```

instead of needing to check for riscv, emscripten, and whatever other systems we can think of.
if they all are going to end up using simde, we should just have a flag for that instead.